e2e/container/attach_test.go

```diff
@@ -32,7 +33,13 @@ func withStdinNewline(cmd *icmd.Cmd) {
 
 // Regression test for https://github.com/docker/cli/issues/5294
 func TestAttachInterrupt(t *testing.T) {
-	result := icmd.RunCommand("docker", "run", "-d", fixtures.AlpineImage, "sh", "-c", "sleep 5")
+	// this is a new test, it already did not work (inside dind) when over ssh
+	// todo(laurazard): make this test work w/ dind over ssh
+	skip.If(t, strings.Contains(os.Getenv("DOCKER_HOST"), "ssh://"))
+
+	// if
+	result := icmd.RunCommand("docker", "run", "-d", fixtures.AlpineImage,
+		"sh", "-c", "trap \"exit 33\" SIGINT; sleep 10; exit 34")
```

Comment:
The `sleep 10` won't be interrupted by the signal, so it will make the test wait for the whole 10s.

We can wait N times for a short intervals to make the signal being handled quicker:
```suggestion
		"sh", "-c", "trap \"exit 33\" SIGINT; for i in $(seq 100); do sleep 0.1; done; exit 34")
```

Reply:
You're right 🤦‍♀️ I initially had it with a `while true; do sleep 1; done; exit x` loop, and should've kept something like that.